src/persist/src/indexed/mod.rs

```diff
@@ -774,6 +789,27 @@ impl<L: Log, B: Blob> Indexed<L, B> {
             return;
         }
 
+        // Check that the request belongs to a stream with currently in-flight
+        // compaction requests. We expect this condition to always be true while
+        // the connection between Indexed and Maintainer is infallible.
+        //
+        // TODO: can we get rid of this check and simply attempt to handle the response
+        // we've received?
+        soft_assert_or_log!(self.in_flight_trace_compactions.contains_key(&id),
```

Comment:
get rid of soft asserts - everything either panics in debug or noop / no logging in prod.

Reply:
also push the "does this arrangement exist" check into `apply_batched_cmd`